### ISSUE TITLE
[Android] Dismissing dialog via back button doesn't cancel Task

### ISSUE THREAD

**Issue by tbalcom** [MvvmCross/MvvmCross#2245]
Here is a sample demonstrating the issue:
https://github.com/tbalcom/MvxDialogBackButton

Here is a StackOverflow question discussing the issue:
https://stackoverflow.com/questions/46456161/mvxappcompatdialogfragment-cancellation-via-back-button-is-not-completely-workin

### Steps to reproduce

1. Click the "Show Dialog" button.

2. Press the Android back button *or* tap outside the dialog to dismiss the dialog.

3. The dialog is closed but the Task used to wrap the navigation is not signaled as done. The "Show Dialog" button remains disabled because it's waiting for the `Task` to finish.

### Expected behavior
The `Task` should be cancelled by the MvvmCross framework if the dialog is dismissed via the Android back button, or the area outside the dialog is tapped to close a cancellable dialog.

### Actual behavior
The `Task` remains active but never completes.

### Configuration

**Version:** 5.2.1

**Platform:** 
- [ ] iOS
- [ x ] Android
- [ ] WPF
- [ ] UWP

**Comment by tbalcom** [comment]
`MvxDialogFragment` could override `OnCancel` and close the ViewModel using the `IMvxNavigationService` if the `MvxDialogFragmentPresentation.Cancelable` is `true`. This proposed change would cause the navigation `Task` to finish if the user cancels the dialog by tapping outside of the dialog area.

**Comment by tbalcom** [comment]
`MvxDialogFragment` could call `Dialog?.SetOnKeyListener(this)` in the `OnResume` override to detect the Android back button key press through `IDialogInterfaceOnKeyListener.OnKey` where `keyCode` is `Keycode.Back`.

**Comment by nmilcoff** [comment]
Thanks for reporting this issue! It was fixed in #2256.

**Comment by tbalcom** [comment]
This bug is not fixed in MvvmCross 5.4.2 if the dialog implements `IMvxViewModelResult<T>` and is navigated to using `await _mvxNavigationService.Navigate<ViewModel, T>();`. The bug is reproducible using the same `MvxDialogBackButton` sample demonstrating the issue in the original post. Tapping the Android back button, or tapping outside of the dialog both result in the `Task` remaining active.

**Comment by nmilcoff** [comment]
@tbalcom I've taken a look at your project and it's true, the Task remains active without changing anything.
But it is because you're only implementing the interface `IMvxViewModelResult<T>` instead of deriving from the implementation `MvxViewModelResult<T>`. 

Changing this line:

```c#
public class NavigationResultLostViewModel : MvxViewModel, IMvxViewModelResult<int?>
```

For this one:

```c#
public class NavigationResultLostViewModel : MvxViewModelResult<int?>
```

Solves the issue.

__Note:__ If you still want to just implement the interface, then this is how your `ViewModel.ViewDestroy` method should look like:

```c#
public override void ViewDestroy()
{
        if(CloseCompletionSource != null && !CloseCompletionSource.Task.IsCompleted && !CloseCompletionSource.Task.IsFaulted)
            CloseCompletionSource?.TrySetCanceled();
        base.ViewDestroy();
}
```
